### ISSUE TITLE
otpauth with unusual timestep

### ISSUE THREAD

**Issue by tycho** [Yubico/yubioath-desktop#90]
One of the keys I want to add to the Yubico Authenticator uses an usual timestep. Instead of the standard 30 seconds, it has a 32-second timestep. The otpauth URI format has provisions for describing the timestep, via the "period" parameter:

https://github.com/google/google-authenticator/wiki/Key-Uri-Format

But the Yubico Authenticator doesn't appear to pay attention to this parameter, which makes it generate invalid TOTP values for this particular use case. Can you please implement support for the "period" parameter? I'd like to consolidate all my OTP keys onto my Yubikey, but this is preventing me from doing so.

**Comment by jas4711** [comment]
Thanks for the report -- I agree it would be a good thing to support (patches welcome).  Out of curiosity, can you mention the system or software that used this?  Always interesting to learn why we are adding a feature.

**Comment by tycho** [comment]
The [Mi-Token android app](https://play.google.com/store/apps/details?id=com.mitoken.otp2&hl=en). It does use a base32-encoded private key, but plugging that into Yubico Authenticator or Google Authenticator wasn't sufficient to replicate the softoken output. I had to decompile the app to figure out what it was doing differently. It turns out it's using the standard RFC6238 algorithm. The only difference is that it uses a 32-second timestep. I tested with `oathtool -s 32` to verify that that's all that was different.
